### PR TITLE
`CloudBuildCreateBuildOperator`: Remove deprecated `body` parameter

### PR DESCRIPTION
**Pull request by eladkal**
* `CloudBuildCreateBuildOperator`: Remove `body`. Please use `build`

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/operators/cloud_build.py [hunk 1]
```diff
@@ -144,8 +142,7 @@ class CloudBuildCreateBuildOperator(BaseOperator):
     def __init__(
         self,
         *,
-        build: Optional[Union[Dict, Build]] = None,
-        body: Optional[Dict] = None,
```
Comment (josh-fell) [inline airflow/providers/google/cloud/operators/cloud_build.py:148]: `body` should be removed from `template_fields` as well.

File: airflow/providers/google/cloud/operators/cloud_build.py [hunk 2]
```diff
@@ -113,10 +112,9 @@ class CloudBuildCreateBuildOperator(BaseOperator):
         For more information on how to use this operator, take a look at the guide:
         :ref:`howto/operator:CloudBuildCreateBuildOperator`
 
-    :param build: Optional, the build resource to create. If a dict is provided, it must be of
+    :param build: The build resource to create. If a dict is provided, it must be of
         the same form as the protobuf message `google.cloud.devtools.cloudbuild_v1.types.Build`.
         Only either build or body should be passed.
```
Comment (josh-fell) [inline airflow/providers/google/cloud/operators/cloud_build.py]:
```suggestion
```
Seems unnecessary now. WDYT?

File: airflow/providers/google/cloud/operators/cloud_build.py [hunk 3]
```diff
@@ -113,10 +112,9 @@ class CloudBuildCreateBuildOperator(BaseOperator):
...
-    :param body: (Deprecated) The build resource to create.
         This parameter has been deprecated. You should pass the build parameter instead.
```
Comment (josh-fell) [inline airflow/providers/google/cloud/operators/cloud_build.py]:
```suggestion
```
Same here.

File: airflow/providers/google/cloud/operators/cloud_build.py [hunk 4]
```diff
@@ -113,10 +112,9 @@ class CloudBuildCreateBuildOperator(BaseOperator):
         For more information on how to use this operator, take a look at the guide:
         :ref:`howto/operator:CloudBuildCreateBuildOperator`
 
-    :param build: Optional, the build resource to create. If a dict is provided, it must be of
+    :param build: The build resource to create. If a dict is provided, it must be of
```
Comment (josh-fell) [inline airflow/providers/google/cloud/operators/cloud_build.py:115]: Looking the `prepare_template()` method, it seems like this value could be also be a path to YAML or JSON file. Is it worth including that in the parameter description too?

Comment (eladkal) [reply airflow/providers/google/cloud/operators/cloud_build.py:115]: There are pointers to some refactoring needed in https://github.com/apache/airflow/pull/18184/files#r706775739
so I prefer to leave it out from this PR. I will open a followup issue on it so we can keep track

Comment (eladkal) [reply airflow/providers/google/cloud/operators/cloud_build.py:115]: FYI @ryanahamilton maybe you want to revisit issue raised in the [comment](https://github.com/apache/airflow/pull/18184/files#r706775739)?

Comment (eladkal) [reply airflow/providers/google/cloud/operators/cloud_build.py:115]: FYI  @ryanyuan maybe you want to revisit issue raised in the [comment](https://github.com/apache/airflow/pull/18184/files#r706775739)?

**Review by josh-fell — APPROVED** [review]

**Pull request merged**